views/js/submitScore.js

```diff
@@ -1,16 +1,22 @@
+document.addEventListener('pacmanLoad', () => {
+  const iframe = document.getElementById("kubercade_iframe");
+  iframe.addEventListener("load", () => {
+    const iframeWindow = iframe.contentWindow;
+    const origGameover = iframeWindow.gameover;
+    iframeWindow.gameover = () => {
+      score = getScore()
+      origGameover();
+      scorePopUp(score);
+    }
+  });
+});
+
 const scorePopUp = () => {
-  const iframeLoc = document.getElementById(
-    "kubercade_iframe").contentWindow.location.href;
-  if (!iframeLoc.endsWith("games/pacman/")) {
-    alert("You can't submit a score if you aren't playing a game!");
-    return;
-  }
   const name = prompt("Please enter your name:", "anonymous");
   // don't post if person cancels prompt or doesn't enter name value
-  if (name == null || name == "") {
+  if (!name) {
     return;
   }
-  const score = getScore();
   sendScore("/scores/pacman/", {
     name,
     score
```

Comment:
Is there a way to make `scorePopUp` generic so it doesn't have variables, elements, and urls specific to Pac-Man hard coded in? Maybe try passing a custom anonymous `getScore()` function and a score submission URL in the `document.addEventListener('pacmanLoad', () => {` iframe event listener block. I just want to make a system that's very modular and easy to extend to other games with as few adaptations and repeated code as possible.

Reply:
Added some parameters to make `scorePopUp` generic, 02ea219